### PR TITLE
created group_series_by_label_names Go-binding

### PR DESCRIPTION
**Pull request by cherep58**

### CODE REVIEW

File: pp/series_index/querier/series_operations.h [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    PRAGMA_DIAGNOSTIC(ignored "-Wunused-local-typedefs")
+    using is_transparent = void;
+    PRAGMA_DIAGNOSTIC(pop)
+
+    [[nodiscard]] PROMPP_ALWAYS_INLINE size_t operator()(const SetItem& key) const noexcept {
+      return BareBones::XXHash3::hash(key.values_ids.data(), key.values_ids.size());
+    }
+
+    [[nodiscard]] PROMPP_ALWAYS_INLINE size_t operator()(const BareBones::Vector<uint32_t>& values_ids) const noexcept {
+      return BareBones::XXHash3::hash(values_ids.data(), values_ids.size());
```
Comment (gshigin) [inline pp/series_index/querier/series_operations.h]: XXHash3::hash expects size in bytes, so there should be `values_ids.size() * sizeof(uint32_t)`?

**Review by gshigin — CHANGES_REQUESTED** [review]

**Review by gshigin — APPROVED** [review]

File: pp/series_index/querier/series_operations.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  class HashCalculator {
+   public:
+    PRAGMA_DIAGNOSTIC(push)
+    PRAGMA_DIAGNOSTIC(ignored "-Wunused-local-typedefs")
+    using is_transparent = void;
+    PRAGMA_DIAGNOSTIC(pop)
+
+    [[nodiscard]] PROMPP_ALWAYS_INLINE size_t operator()(const SetItem& key) const noexcept {
+      return BareBones::XXHash3::hash(key.values_ids.data(), key.values_ids.size());
```
Comment (vporoshok) [inline pp/series_index/querier/series_operations.h]: This overload still hashes only `values_ids.size()` bytes, while the transparent lookup overload below hashes the whole `uint32_t` span (`size() * sizeof(uint32_t)`). Since `EqualTo` says `SetItem` and `BareBones::Vector<uint32_t>` are equal when their vectors match, `phmap` requires both hash overloads to return the same value for equal keys. After the set grows/rehashes, an existing group inserted with this hash can become unreachable from lookups that use the span-based hash, so equal label tuples may be emitted as separate groups. Please use the same span-based hash here too, and ideally add a test where a duplicate group appears after many unique groups.

File: pp/go/cppbridge/lss_snapshot.go [hunk 3]
```diff
@@ -77,6 +77,17 @@ func (lss *LabelSetSnapshot) Query(selector uintptr) *LSSQueryResult {
 	return result
 }
 
+// GroupSeriesByLabelNames group series by label names
+func (lss *LabelSetSnapshot) GroupSeriesByLabelNames(seriesIDs []uint32, labelNameIDs []uint32) [][]uint32 {
```
Comment (vporoshok) [inline pp/go/cppbridge/lss_snapshot.go]: This returns a raw `[][]uint32` backed by C++ allocation, and correctness now depends on every caller remembering to call `FreeGroupedSeries`. Other C-allocated results in this package are wrapped with finalizers, so this public API is easy to leak; the tests would also leak if an assertion fails before the explicit free. Could we return a small result wrapper with a finalizer, or otherwise make the ownership explicit and hard to miss?

**Review by vporoshok — CHANGES_REQUESTED** [review]
Thanks for the update. I found two issues that should be addressed before merging: one correctness bug in the grouping hash and one ownership issue in the new Go API.

**Review by vporoshok — APPROVED** [review]
